public/app/features/plugins/extensions/getPluginExtensions.test.ts

```diff
@@ -0,0 +1,213 @@
+  const placement2 = 'plugins/myorg-basic-app/start';
+  const pluginId = 'grafana-basic-app';
+  let link1: PluginExtensionLinkConfig, link2: PluginExtensionLinkConfig;
+
+  beforeEach(() => {
+    link1 = {
+      title: 'Link 1',
+      description: 'Link 1 description',
+      path: `/a/${pluginId}/declare-incident`,
+      placement: placement1,
+      configure: jest.fn().mockReturnValue({}),
+    };
+    link2 = {
+      title: 'Link 2',
+      description: 'Link 2 description',
+      path: `/a/${pluginId}/declare-incident`,
+      placement: placement2,
+      configure: jest.fn().mockImplementation((context) => ({ title: context?.title })),
+    };
+
+    global.console.warn = jest.fn();
+  });
+
+  test('should return the extensions for the given placement', () => {
+    const registry = createPluginExtensionRegistry([{ pluginId, linkExtensions: [link1, link2] }]);
+    const { extensions } = getPluginExtensions({ registry, placement: placement1 });
+
+    expect(extensions).toHaveLength(1);
+    expect(extensions[0]).toEqual(
+      expect.objectContaining({
+        pluginId,
+        type: PluginExtensionTypes.link,
+        title: link1.title,
+        description: link1.description,
+        path: link1.path,
+      })
+    );
+  });
+
+  test('should return with an empty list if there are no extensions registered for a placement yet', () => {
+    const registry = createPluginExtensionRegistry([{ pluginId, linkExtensions: [link1, link2] }]);
+    const { extensions } = getPluginExtensions({ registry, placement: 'placement-with-no-extensions' });
+
+    expect(extensions).toEqual([]);
+  });
+
+  test('should pass the context to the configure() function', () => {
+    const context = { title: 'New title from the context!' };
+    const registry = createPluginExtensionRegistry([{ pluginId, linkExtensions: [link2] }]);
+
+    getPluginExtensions({ registry, context, placement: placement2 });
+
+    expect(link2.configure).toHaveBeenCalledTimes(1);
+    expect(link2.configure).toHaveBeenCalledWith(context);
+  });
+
+  test('should be possible to update the basic properties with the configure() function', () => {
+    link2.configure = jest.fn().mockImplementation(() => ({
+      title: 'Updated title',
+      description: 'Updated description',
+      path: `/a/${pluginId}/updated-path`,
+    }));
+
+    const registry = createPluginExtensionRegistry([{ pluginId, linkExtensions: [link2] }]);
+    const { extensions } = getPluginExtensions({ registry, placement: placement2 });
+    const [extension] = extensions;
+
+    assertPluginExtensionLink(extension);
+
+    expect(link2.configure).toHaveBeenCalledTimes(1);
+    expect(extension.title).toBe('Updated title');
+    expect(extension.description).toBe('Updated description');
+    expect(extension.path).toBe(`/a/${pluginId}/updated-path`);
+  });
+
+  test('should not be possible to override certain properties with the configure() function', () => {
```

Comment:
Maybe in this scenario we should hide the extension instead of fallback to the old values. My thinking is that we want to reduce the risk of sending the user to a path/place that doesn't make sense.

Reply:
Fair enough, let's do it like this 👍